cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ListenersValidator.java

```diff
@@ -560,4 +564,20 @@ private static void validateOauth(Set<String> errors, GenericKafkaListener liste
     private static List<String> getNames(List<GenericKafkaListener> listeners)    {
         return listeners.stream().map(GenericKafkaListener::getName).distinct().collect(Collectors.toList());
     }
+
+
+    /**
+     * Filters configuration provided in any custom authentication listeners to remove any forbidden properties
+     * @param reconciliation Current reconciliation attempt
+     * @param listener Listener to filter
+     */
+    private static void filterCustomAuthConfiguration(Reconciliation reconciliation, GenericKafkaListener listener) {
+        if (isListenerWithCustomAuth(listener)) {
+            KafkaListenerAuthenticationCustom customAuth = (KafkaListenerAuthenticationCustom) listener.getAuth();
+            if (customAuth.getListenerConfig() != null) {
+                KafkaListenerCustomAuthConfiguration config = new KafkaListenerCustomAuthConfiguration(reconciliation, customAuth.getListenerConfig().entrySet());
+                customAuth.setListenerConfig(Map.copyOf(config.asOrderedProperties().asMap()));
```

Comment:
I think modifying the custom resource is fairly dangerous. Can't this be used instead in `KafkaBrokerConfigurationBuilder.java` when passing the configuration to the broker config file? There you can remove the forbidden options without modifying the resource and storing the anywhere.

Reply:
@scholzj the only issue is that the `Reconciliation` object has no reference here, which is required for the AbstractConfig class. I'm happy to make a constructor that doesn't require that though, or simply pass in the `Reconciliation` object to builder alternatively?

Reply:
I doubt you can easily create a constructor without the reconciliation parameter because it is passed to the parent constructor. So I guess passing the reconciliation to the `KafkaBrokerConfigurationBuilder` class, probably in its constructor might be the easier way.